Sources/MoEngage-Swift-Segment/MoEngage_Swift_Segment.swift

```diff
@@ -0,0 +1,240 @@
+                }
+            }
+                        
+            if let phone = traits[UserAttributes.phone.rawValue] as? String {
+                MoEngageSDKAnalytics.sharedInstance.setMobileNumber(phone, forAppID: moengageSettings?.apiKey)
+            }
+       
+            if let isoDate = traits[UserAttributes.isoDate.rawValue] as? [String: Any] {
+                if let date = isoDate["date"] as? Date, let attributeName = isoDate["attributeName"] as? String {
+                    MoEngageSDKAnalytics.sharedInstance.setUserAttributeDate(date, withAttributeName: attributeName, forAppID: moengageSettings?.apiKey)
+                }
+            }
+            
+            if let location = traits[UserAttributes.location.rawValue] as? [String: Any] {
+                if let latitute = location["latitude"] as? Double, let longitude = location["longitude"] as? Double {
+                    MoEngageSDKAnalytics.sharedInstance.setLocation(MoEngageGeoLocation.init(withLatitude: latitute, andLongitude: longitude))
+                }
+            }
+            
+            let moengageTraits = UserAttributes.allCases.map { $0.rawValue }
+            
+            for trait in traits where !moengageTraits.contains(trait.key) {
+                switch trait.value {
+                case let val as String:
+                    MoEngageSDKAnalytics.sharedInstance.setUserAttribute(val, withAttributeName: trait.key, forAppID: moengageSettings?.apiKey)
+                case let val as Date:
+                    MoEngageSDKAnalytics.sharedInstance.setUserAttributeDate(val, withAttributeName: trait.key, forAppID: moengageSettings?.apiKey)
+                case let val as Bool:
+                    MoEngageSDKAnalytics.sharedInstance.setUserAttribute(val, withAttributeName: trait.key, forAppID: moengageSettings?.apiKey)
+                case let val as Int:
+                    MoEngageSDKAnalytics.sharedInstance.setUserAttribute(val, withAttributeName: trait.key, forAppID: moengageSettings?.apiKey)
+                case let val as Double:
+                    MoEngageSDKAnalytics.sharedInstance.setUserAttribute(val, withAttributeName: trait.key, forAppID: moengageSettings?.apiKey)
+                default:
+                    MoEngageSDKAnalytics.sharedInstance.setUserAttribute(trait.value, withAttributeName: trait.key, forAppID: moengageSettings?.apiKey)
+                }
+            }
+        }
+
+        return event
+    }
+
+    public func track(event: TrackEvent) -> TrackEvent? {
+        if var generalAttributeDict = event.properties?.dictionaryValue {
+            var dateAttributeDict: [AnyHashable : Any] = [:]
+            
+            for key in generalAttributeDict.keys {
+                let val = generalAttributeDict[key]
+                if val == nil || (val as? NSNull) == NSNull() {
+                    generalAttributeDict.removeValue(forKey: key)
```

Comment:
@deepa-iOS-SDK can u check if its handled in native.. if yes it can be skipped

Reply:
@RakshithaAcharya This logic is from the previous version of MoEngageSegment and I see that if the value is nil we remove generalAttributeDict value and then add generalAttributeDict for MoEngageProperties. If we skip the data being attached will be different. let me know if this should be done here